### PR TITLE
Finalize integration calculations and clean output formatting

### PR DESCRIPTION
**Pull request by dnxjay**
For this assignment all pull requests going  forward should contain the following (please delete the placeholder text and add your own description)

# Title
  - Calculate Integrals for Given Functions

## Linting
  - The code was run through pylint, and there are no critical linting issues. Minor warnings related to formatting were ignored as they do not impact functionality.

## Contents
  -This pull request implements the integration of three given functions over specified boundary using Simpson's Rule. The results are printed to the console, and the output includes the computed integral values and the number of steps (where applicable).

## Reviewer
  - @Cosmos491

### CODE REVIEW

**Comment by dnxjay** [comment]
@Cosmos491 this PR is ready to be reviewed.

**Comment by zbpetersbuf** [comment]
@Cosmos491 please go ahead with this review

**Comment by dnxjay** [comment]
> @Cosmos491 please go ahead with this review

can we potentially assign a different reviewer if not done by tomorrow?

**Comment by zbpetersbuf** [comment]
@dnxjay ya I was thinking about that aswell is there a time you want to add another reviewer? I was thinking late afternoon Sunday but it could always be sooner.
If you are concerned about this not getting this merged, I can always force merge it on Monday night  to make sure you get the points for it (but you would need to make sure that the branch is updated before I do)

**Comment by zbpetersbuf** [comment]
@dnxjay @Cosmos491, dnxjay needs to get this pull request merged so that they can merge their next one, so it needs a reviewer to do this, if you do not review this by some time tomorrow afternoon there will be a new review assigned

**Comment by dnxjay** [comment]
@laserlab are you able to explain this linting error? Im not quiet sure where im going wrong. I also keep passing 10/10 in codespace but not in this PR so im confused about that as well.

**Comment by Cosmos491** [comment]
For the listing error that is being shown here: 

Could there be any mismatch Between Function Definition and Function Call: Can you check function simpson() calculus.py and see how many arguments it can accept. It seems like you are passing 4 arguments (f, a, b, 1000) to the function, but the function definition expects fewer arguments ? @dnxjay

**Comment by laserlab** [comment]
> For the listing error that is being shown here: 
> 
> 
> 
> Could there be any mismatch Between Function Definition and Function Call: Can you check function simpson() calculus.py and see how many arguments it can accept. It seems like you are passing 4 arguments (f, a, b, 1000) to the function, but the function definition expects fewer arguments ? @dnxjay 

You are using Simpson from the active namespace for which the error makes sense. 

Use the wrapper that was written for it instead.

**Comment by dnxjay** [comment]
@zbpetersbuf we might need to add another reviewer for this

**Comment by zbpetersbuf** [comment]
alright if your sure, i will do that

**Comment by zbpetersbuf** [comment]
@avgagliardo @kylemasc917 would one of you review this pull request

**Comment by avgagliardo** [comment]
I can review this. I start it now.

File: calculus.py
Comment (avgagliardo) [inline calculus.py:1]: Code is well-formed, includes necessary docstrings and code comments.

Execution proceeds as expected.

Passes all units tests, and lints with a score of 10/10.

**Review by avgagliardo — COMMENTED** [review]
Code is well-formed, includes necessary docstrings and code comments.

Execution proceeds as expected.

Passes all units tests, and lints with a score of 10/10.

Approving for merge.

**Review by zbpetersbuf — APPROVED** [review]

**Comment by zbpetersbuf** [comment]
@avgagliardo you said you approved the merge but didnt select approve, this means that i will have to force merge which i can do, but you will not get the points for this if i do
Screenshot 2024-11-24 at 5 49 32 PM

**Comment by Cosmos491** [comment]
> only one of you needs to review, @Cosmos491 this is your last chance

But I did leave my review.

**Review by avgagliardo — APPROVED** [review]
Approved for merge.

**Comment by avgagliardo** [comment]
Woops, I see what you mean. Sorry about that! Should be fixed now.

**Comment by zbpetersbuf** [comment]
@Cosmos491 you left a review but never approved it, approving it is when you get the points, want to approve it now?

**Comment by zbpetersbuf** [comment]
@Cosmos491 you still got time since @dnxjay still needs to update their branch

**Comment by dnxjay** [comment]
> @Cosmos491 you still got time since @dnxjay still needs to update their branch

branch should be updated and ready for merge

**Review by Cosmos491 — APPROVED** [review]

**Comment by zbpetersbuf** [comment]
i said id give @Cosmos491 10 min a min ago if thats alright

**Comment by zbpetersbuf** [comment]
o cool ill merge it then

**Pull request merged**
